Cython/Utility/Coroutine.c

```diff
@@ -1,9 +1,31 @@
+//////////////////// CoroutineSetYieldFrom ////////////////////
+
+static void
+__Pyx_Coroutine_Set_Owned_Yield_From(__pyx_CoroutineObject *gen, PyObject *yf) {
+    // NOTE: steals a reference to yf by transfering it to 'gen->yieldfrom' !
+    assert (!gen->yieldfrom_am_send);
+    assert (!gen->yieldfrom);
+#if PY_VERSION_HEX < 0x030A00F0
+    if (__Pyx_PyType_HasFeature(Py_TYPE(yf), __Pyx_TPFLAGS_HAVE_AM_SEND))
+#endif
+    {
+        __Pyx_PyAsyncMethodsStruct* am = __Pyx_PyType_AsAsync(yf);
+        if (likely(am)) {
+            // Keep a direct reference to am->am_send, if provided.
+            gen->yieldfrom_am_send = am->am_send;
```

Comment:
Would be good if this could work/at least not break compilation with Limited API

I think it's fine for 3.10+ (we can just get it with the normal slot mechanism) but below that we doing have access to `__Pyx_PyAsyncMethodsStruct` (so it just has to be skipped).

Reply:
Right. I think a `TryGetSlot()` protected by `! LIMITED || LIMITED_VERSION_HEX >= 3.10` should make this work (or disable it) in both settings.